src/coreclr/inc/corinfo.h

```diff
@@ -438,7 +438,7 @@ enum CorInfoHelpFunc
     CORINFO_HELP_CHKCASTCLASS_SPECIAL, // Optimized helper for classes. Assumes that the trivial cases
                                     // has been taken care of by the inlined check
 
-    CORINFO_HELP_BOX,
+    CORINFO_HELP_BOX,               // Fast box helper. Only possible exception is OutOfMemory
```

Comment:
This comment is in conflict with the implementation that checks and throws for IsValueType.

Reply:
Delete the check in the implementation and replace it with assert?

Reply:
And/or move it to `CEEInfo::getBoxHelper`

Reply:
Ah ok, I have noticed that you have done it already :-)

Reply:
I removed the check for `IsValueType`.